android/src/main/java/com/dylanvann/fastimage/FastImageViewWithUrl.java

```diff
@@ -1,14 +1,129 @@
+        if (!mNeedsReload)
+            return;
+
+        if ((mSource == null ||
+                !mSource.hasKey("uri") ||
+                isNullOrEmpty(mSource.getString("uri"))) &&
+                mDefaultSource == null) {
+
+            // Cancel existing requests.
+            if (requestManager != null) {
+                requestManager.clear(this);
+            }
+
+            if (glideUrl != null) {
+                FastImageOkHttpProgressGlideModule.forget(glideUrl.toStringUrl());
+            }
+
+            // Clear the image.
+            setImageDrawable(null);
+            return;
+        }
+
+        //final GlideUrl glideUrl = FastImageViewConverter.getGlideUrl(view.getContext(), mSource);
+        final FastImageSource imageSource = FastImageViewConverter.getImageSource(getContext(), mSource);
+        final GlideUrl glideUrl = imageSource == null ? null : imageSource.getGlideUrl();
+
+        // Cancel existing request.
+        this.glideUrl = glideUrl;
+        if (requestManager != null) {
+            requestManager.clear(this);
+        }
+
+        String key = glideUrl == null ? null : glideUrl.toStringUrl();
+
+        if (glideUrl != null) {
+            FastImageOkHttpProgressGlideModule.expect(key, manager);
+            List<FastImageViewWithUrl> viewsForKey = viewsForUrlsMap.get(key);
+            if (viewsForKey != null && !viewsForKey.contains(this)) {
+                viewsForKey.add(this);
+            } else if (viewsForKey == null) {
+                List<FastImageViewWithUrl> newViewsForKeys = new ArrayList<>(Collections.singletonList(this));
+                viewsForUrlsMap.put(key, newViewsForKeys);
+            }
+        }
+
+        ThemedReactContext context = (ThemedReactContext)getContext();
+        if (imageSource != null) {
+            // This is an orphan even without a load/loadend when only loading a placeholder
+
+            RCTEventEmitter eventEmitter = context.getJSModule(RCTEventEmitter.class);
+            int viewId = this.getId();
+
+            eventEmitter.receiveEvent(viewId,
+                    FastImageViewManager.REACT_ON_LOAD_START_EVENT,
+                    new WritableNativeMap());
+        }
+
+        if (requestManager != null) {
+            RequestBuilder<Drawable> builder =
+            requestManager
+                    // This will make this work for remote and local images. e.g.
+                    //    - file:///
+                    //    - content://
+                    //    - res:/
+                    //    - android.resource://
+                    //    - data:image/png;base64
+                    .load(imageSource == null ? null : imageSource.getSourceForLoad())
+                    .placeholder(mDefaultSource) // show until loaded
+                    .fallback(mDefaultSource) // null will not be treated as error
+                    .apply(FastImageViewConverter.getOptions(context, imageSource, mSource));
```

Comment:
@danielgindi the local defaultSource image was not showing (tested on multiple emulators/devices and RN 0.61.5/0.62.2). Had to make following changes to make it work. Could you take a look?

```suggestion
                    .apply(FastImageViewConverter
                            .getOptions(context, imageSource, mSource)
                            .placeholder(mDefaultSource) // show until loaded
                            .fallback(mDefaultSource)); // null will not be treated as error              
```

Reply:
Could it be due to an update to the underlying library?
I have not seen such an issue, will have to test...

Reply:
I have had same issue on Android. 
Suggestion made by @vinayr fixed the problem for me as well :) 
According to SO [link](https://stackoverflow.com/questions/45481756/glide-4-0-0-missing-placeholder-error-glideapp-and-does-not-resolve-its-method) & [link](https://stackoverflow.com/questions/47262017/glide-4-3-1-override-and-placeholder-features-not-work) this is @vinayr's suggestion is way to go!

Reply:
@danielgindi I have just tested your branch in real device for both iOS and Android, so turns out for defaultSource in iOS is totally working, but in Android device seems not. So if the Glide api changed maybe should update it.